src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1025,10 +1016,95 @@ private void CheckNewModifier(Symbol symbol, bool isNew, DiagnosticBag diagnosti
                 suppressAccessors = true;
             }
 
+            return;
+
             bool isOrContainsErrorType(TypeSymbol typeSymbol)
             {
                 return (object)typeSymbol.VisitType((currentTypeSymbol, unused1, unused2) => currentTypeSymbol.IsErrorType(), (object)null) != null;
             }
+
+            static void checkValidNullableMethodOverride(
+                Location overridingMemberLocation,
+                MethodSymbol overriddenMethod,
+                MethodSymbol overridingMethod,
+                DiagnosticBag diagnostics,
+                bool checkParameters)
+            {
+                var compilation = overridingMethod.DeclaringCompilation;
+                if (overriddenMethod is null ||
+                    overridingMethod is null ||
+                    compilation is null ||
+                    ((CSharpParseOptions)overridingMemberLocation.SourceTree?.Options)
+                        ?.IsFeatureEnabled(MessageID.IDS_FeatureNullableReferenceTypes) != true)
+                {
+                    // Don't do any validation if the nullable feature is not enabled or
+                    // the override is not written directly in source
+                    return;
+                }
+
+                var conversions = compilation.Conversions.WithNullability(true);
+                if (!isValidNullableConversion(
+                        conversions,
+                        overridingMethod.RefKind,
+                        overridingMethod.ReturnTypeWithAnnotations,
```

Comment:
>overridingMethod.ReturnTypeWithAnnotations [](start = 24, length = 42)

Has one of these been alpha-renamed to match the other, e.g. in case the method is generic and returns `T`?